### PR TITLE
Use a modified config drive so we don't set ssh keys

### PR DESCRIPTION
**Pull request by aweeks**

### CODE REVIEW

File: lib/finalize_rescue.bash [hunk 1]
```diff
@@ -1,31 +1,66 @@
...
+
+# Prepare new config drive
+mkdir -p ${cdWorkdir}/openstack/latest
+mkdir -p ${cdWorkdir}/openstack/content
+
+# Copy all content (specificaly the interfaces file)
+cp ${cdMountpoint}/openstack/content/* ${cdWorkdir}/openstack/content/
+
+# Select only the cloud config drive fields we want
+jq '{network_config, name, hostname, uuid}' ${cdMountpoint}/openstack/latest/meta_data.json > ${cdWorkdir}/openstack/latest/meta_data.json
```
Comment (jimrollenhagen) [inline lib/finalize_rescue.bash]: network configuration is in vendor_data.json, fwiw.

Comment (jimrollenhagen) [inline lib/finalize_rescue.bash]: I take that back, we're using debian-style here.

File: lib/finalize_rescue.bash [hunk 2]
```diff
@@ -1,31 +1,66 @@
...
 mount -o loop ${cdIsoFile} ${cdMountpoint}
-coreos-cloudinit --from-configdrive=${cdMountpoint} --convert-netconf="debian"
-umount ${cdMountpoint}
+
+# Prepare new config drive
+mkdir -p ${cdWorkdir}/openstack/latest
+mkdir -p ${cdWorkdir}/openstack/content
+
+# Copy all content (specificaly the interfaces file)
+cp ${cdMountpoint}/openstack/content/* ${cdWorkdir}/openstack/content/
```
Comment (pquerna) [inline lib/finalize_rescue.bash]: just use a `cp -r` or `rsync` ?

Comment (pquerna) [inline lib/finalize_rescue.bash]: (i dislike `*` shell expansions in scripts)

File: lib/finalize_rescue.bash [hunk 3]
```diff
@@ -1,31 +1,65 @@
...
-rescueHash=${3}
+PATH="${PATH}:/usr/share/oem/bin"
+
+rescueUsername=${1}
+rescueHash=${2}
+clean="true"
+if [ -n "${3}" ]
+then
+    clean="false"
+fi
```
Comment (jimrollenhagen) [inline lib/finalize_rescue.bash]: What's $3 for? Don't see it used.

Comment (jimrollenhagen) [inline lib/finalize_rescue.bash]: By used, I mean ever passed in. Any reason for this code?

Comment (aweeks) [inline lib/finalize_rescue.bash]: Could be better documented, but basically a debug flag that doesn't unmount/delete temp files.

File: lib/finalize_rescue.bash [hunk 4]
```diff
@@ -1,31 +1,65 @@
...
+# Prepare new config drive
+mkdir -p ${cdWorkdir}/openstack/latest
+
+# Copy all content (specificaly the interfaces file)
+cp -r ${cdMountpoint}/openstack/content ${cdWorkdir}/openstack
+
+# Select only the cloud config drive fields we want
+jq '{network_config, name, hostname, uuid}' ${cdMountpoint}/openstack/latest/meta_data.json > ${cdWorkdir}/openstack/latest/meta_data.json
+
+if [ $clean == "true" ]
```
Comment (jimrollenhagen) [inline lib/finalize_rescue.bash]: I don't know bash insanely well, but isn't [[ double bracket ]] preferred?

**Comment by jimrollenhagen** [comment]
Mostly looks fine to me, a couple things inline, nbd. +1

**Pull request merged**